hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -249,6 +249,19 @@
           + "record size estimate compute dynamically based on commit metadata. "
           + " This is critical in computing the insert parallelism and bin-packing inserts into small files.");
 
+  public static final ConfigProperty<String> MAX_ARCHIVE_FILES_TO_KEEP_PROP = ConfigProperty
+      .key("hoodie.max.archive.files")
+      .defaultValue("10")
+      .withDocumentation("The numbers of kept archive files under archived.");
+
+  public static final ConfigProperty<String> AUTO_TRIM_ARCHIVE_FILES_DROP = ConfigProperty
+      .key("hoodie.auto.trim.archive.files")
+      .defaultValue("false")
+      .withDocumentation("When enabled, Hoodie will keep the most recent " + MAX_ARCHIVE_FILES_TO_KEEP_PROP.key()
```

Comment:
Let's add a `WARNING` in both configs.  sth like : `WARNING: do not use this config unless you know what you're doing.  If enabled, details of older archived instants are deleted, resulting in information loss in the archived timeline, which may affect tools like CLI and repair.  Only enable this if you hit severe performance issues for retrieving archived timeline.` (feel free to add more details)

Reply:
Appreciate it. Changed